### ISSUE TITLE
[Bug] Monitor does not take into account modification introduced by a wrapper.

### ISSUE THREAD

**Issue by qgallouedec** [DLR-RM/stable-baselines3#798]
### 🐛 Bug

When using [`make_vec_env`](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/env_util.py#L38-L98) with `wrapper_class` argument, the monitor does not take into account the modification introduced by the wrapper.

In some way related with #477

### To Reproduce

This is a simple example with a wrapper that remove the reward. But you can encounter the same issue with wrapper that change `done`.

```python
import gym
from stable_baselines3 import SAC
from stable_baselines3.common.env_util import make_vec_env


class MyWrapper(gym.RewardWrapper):
    """A wrapper that remove reward."""

    def reward(self, reward):
        return 0.0


env = make_vec_env("Pendulum-v0", wrapper_class=MyWrapper)
model = SAC("MlpPolicy", env, verbose=1)
model.learn(1000)
```

```bash
Using cpu device
----------------------------------
| rollout/           |           |
|    ep_len_mean     | 200       |
|    ep_rew_mean     | -1.44e+03 |
| time/              |           |
|    episodes        | 4         |
|    fps             | 77        |
|    time_elapsed    | 10        |
|    total_timesteps | 800       |
| train/             |           |
|    actor_loss      | -2.77     |
|    critic_loss     | 0.00882   |
|    ent_coef        | 0.811     |
|    ent_coef_loss   | -0.352    |
|    learning_rate   | 0.0003    |
|    n_updates       | 699       |
----------------------------------

```

### Expected behavior

```bash
|    ep_rew_mean     | 0.0     |
```

### System info

- OS: macOS-12.2.1-x86_64-i386-64bit Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64
- Python: 3.8.12
- Stable-Baselines3: 1.4.1a3
- PyTorch: 1.10.2
- GPU Enabled: False
- Numpy: 1.22.2
- Gym: 0.19.0

### Additional context

I think that you can solve this issue just by wrapping Monitor after the `wrapper_class`.
Let me know if you want me to open a PR.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by Miffyli** [comment]
This is somewhat by design. In the past, Monitor logged rewards based on the modified reward, which is likely not used to measure the actual performance of the agent (e.g. DQN is trained on clipped [-1, 1] reward but evaluated using original reward of the Atari game). I am not 100% sure if ´wrapper_class` should be applied before Monitor for this reason, and at this point could silently change results of code-bases using `wrapper_class` argument.

@araffin thoughts?

**Comment by araffin** [comment]
> So is it relevant to make this change to make_vec_env?

I would not change `make_vec_env` but rather do:
```python
def my_wrappers(env):
    return Monitor(MyWrapper(env))

env = make_vec_env("Pendulum-v0", wrapper_class=my_wrappers)
```

or as you suggest (don't use lambda here, it may have weird effects):
```python
def env_func():
    return MyWrapper(MyEnv())

env = make_vec_env(env_func)
```
